### ISSUE TITLE
Removing badly shaped triangles in CDT

### ISSUE THREAD

**Issue by zinuok** [artem-ogre/CDT#63]
Hello, I successfully created CDT using the following methods, but there are many bad triangles in result (flat triangle), 
which cause an error " In initialize: Triangulation is invalid" when I use "LinearTriInterpolator" method from python matploblib library.

So, my question is, how can I remove the bad shape triangles in CDT?

```
cdt.insertVertices(vertices);
cdt.insertEdges(edges);
CDT::RemoveDuplicatesAndRemapEdges(vertices, edges); // remove duplicate
cdt.eraseSuperTriangle(); 
```

![image](https://user-images.githubusercontent.com/45934290/143880136-b0f0cf56-911d-480a-95d1-506ea67e029b.png)

**Comment by artem-ogre** [comment]
One way to improve such triangles is perform re-meshing by inserting additional points (e.g., Steiner points).
CDT does not provide functionality to do this. So either you need to use another software for doing that, write your own re-meshing or even contribute Steiner point insertion to CDT :)

I am marking this issue as a duplicate of #7 .

**Comment by artem-ogre** [comment]
Sorry for posting in the old thread. Ruppert refinement finally landed in: #225
